### PR TITLE
Add condition for binary classification

### PR DESCRIPTION
**Pull request by GentleZhu**
*Issue #, if available:*
https://github.com/awslabs/graphstorm/issues/541

*Description of changes:*
Add conditions on ROC_AUC and PRAUC calculation to check whether pred_proba only has two classes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/eval/eval_func.py [hunk 1]
```diff
@@ -195,13 +195,17 @@ def eval_roc_auc(logits,labels):
     # The roc_auc_score function computes the area under the receiver operating characteristic
     # (ROC) curve, which is also denoted by AUC or AUROC. The following returns the average AUC.
     rocauc_list = []
-    labels=labels_to_one_hot(labels, predicted_labels.shape[1])
```
Comment (classicsong) [inline python/graphstorm/eval/eval_func.py:198]: Can you add some unit test for this function?

File: python/graphstorm/eval/eval_func.py [hunk 2]
```diff
@@ -195,13 +195,17 @@ def eval_roc_auc(logits,labels):
...
-    for i in range(labels.shape[1]):
-        # AUC is only defined when there is at least one positive data.
-        if np.sum(labels[:, i] == 1) > 0 and np.sum(labels[:, i] == 0) > 0:
-            is_labeled = labels[:, i] == labels[:, i]
-            rocauc_list.append(roc_auc_score(labels[is_labeled, i],
-                                             predicted_labels[is_labeled, i]))
+    if predicted_labels.shape[1] == 2:
```
Comment (classicsong) [inline python/graphstorm/eval/eval_func.py:198]: Can you add comments about in which case `predicted_labels.shape[1] == 2` ?
Do you mean this is binary classification?

File: python/graphstorm/eval/eval_func.py (hunk 1 above)
Comment (GentleZhu) [reply python/graphstorm/eval/eval_func.py:198]: Do we have any existing unit test for binary classification?

**Comment by zhjwy9343** [comment]
In the original eval_func.py line 290, there is another use of auc_roc_score() function. Please also check if this will be impacted by binary classifcation.

**Pull request closed without merging**